libopenage/gui/minimap.cpp

```diff
@@ -293,15 +293,15 @@ std::unique_ptr<QSGNode> Minimap::create_tree() {
 		foreground->markDirty(QSGNode::DirtyGeometry);
 	};
 
-    this->mouse_move_camera = [this, world_to_background, turn] (QMouseEvent *mouseEvent) {
-        const QMatrix4x4 local_to_world_transform = (turn->matrix() * world_to_background->matrix()).inverted();
-        const QPointF pos = local_to_world_transform.map(mouseEvent->localPos());
+	this->mouse_move_camera = [this, world_to_background, turn] (QMouseEvent *mouseEvent) {
+		const QMatrix4x4 local_to_world_transform = (turn->matrix() * world_to_background->matrix()).inverted();
+		const QPointF pos = local_to_world_transform.map(mouseEvent->localPos());
 
-        if (Engine *engine = unwrap(game->get_engine())) {
-            engine->get_coord_data()->camgame_phys.ne = pos.x();
-            engine->get_coord_data()->camgame_phys.se = pos.y();
-        }
-    };
+		if (Engine *engine = unwrap(game->get_engine())) {
+			engine->get_coord_data()->camgame_phys.ne = pos.x();
```

Comment:
This executes in the GUI thread and touches values from the gamelogic thread. (these threads are currently all in one because of the font library that should be removed in the future)

Also, a game may have no engine attached yet (or the game may be null itself).

I'd use something like this:

	if (this->game)
		if (EngineLink *engine = game->get_engine()) {
			emit this->game->in_game_logic_thread_blocking([this, engine] {
				...

Or do all the coordinate changes in the `Minimap::updatePaintNode()`.